gpt_index/indices/query/base.py

```diff
@@ -133,35 +122,22 @@ def __init__(
         # TODO: deprecated
         similarity_cutoff: Optional[float] = None,
         use_async: bool = True,
-        recursive: bool = False,
         streaming: bool = False,
         doc_ids: Optional[List[str]] = None,
         optimizer: Optional[BaseTokenUsageOptimizer] = None,
         node_postprocessors: Optional[List[BaseNodePostprocessor]] = None,
-        llama_logger: Optional[LlamaLogger] = None,
         verbose: bool = False,
     ) -> None:
         """Initialize with parameters."""
         if index_struct is None:
             raise ValueError("index_struct must be provided.")
         self._validate_index_struct(index_struct)
         self._index_struct = index_struct
-        self._llm_predictor = llm_predictor or LLMPredictor()
-        # NOTE: the embed_model isn't used in all indices
-        self._embed_model = embed_model or OpenAIEmbedding()
         if docstore is None:
             raise ValueError("docstore must be provided.")
         self._docstore = docstore
-        self._query_runner = query_runner
-        # TODO: make this a required param
-        if prompt_helper is None:
-            raise ValueError("prompt_helper must be provided.")
-        self._prompt_helper = cast(PromptHelper, prompt_helper)
+        self._service_context = service_context
```

Comment:
if service_context is None, would things break? e.g. how would you call llmpredictor

Reply:
You are right, it should not be Optional.